docs/_posts/examples/3400-01-15-animate-a-line.html

```diff
@@ -0,0 +1,124 @@
+---
+layout: example
+category: example
+title: Animate a line
+description: Animate a line by updating a GeoJSON source on each frame.
+tags:
+- layers
+- sources
+---
+<style>
+button {
+    position: absolute;
+    margin: 20px;
+}
+
+#pause::after {
+    content: 'Pause';
+}
+
+#pause.pause::after {
+    content: 'Play';
+}
+</style>
+<div id='map'></div>
+<button id='pause'></button>
+<script>
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v10',
+    center: [0, 0],
+    zoom: 0.5
+});
+
+// Create a GeoJSON source with an empty lineString.
+var geojson = {
+    "type": "FeatureCollection",
+    "features": [{
+        "type": "Feature",
+        "geometry": {
+            "type": "LineString",
+            "coordinates": [
+                [0, 0]
+            ]
+        }
+    }]
+};
+
+var speedFactor = 30; // number of frames per longitude degree
+var animation;
+var startTime = 0;
+var progress = 0; // progress = timestamp - startTime
+var isActive = false; // indicator of whether the animation needs to be active
+var pauseButton = document.getElementById('pause');
+
+map.on('load', function() {
+
+    // Add a source and layer displaying the line which will be
+    map.addLayer({
+        'id': 'line-animation',
+        'type': 'line',
+        'source': {
+            'type': 'geojson',
+            'data': geojson
+        },
+        'layout': {
+            'line-cap': 'round',
+            'line-join': 'round'
+        },
+        'paint': {
+            'line-color': '#ed6498',
+            'line-width': 5,
+            'line-opacity': .8
+        }
+    });
+
+    startTime = performance.now();
+
+    animateLine();
+
+    pauseButton.addEventListener('click', function() {
+        pauseButton.classList.toggle('pause');
+        if (pauseButton.classList.contains('pause')) {
+            cancelAnimationFrame(animation);
+        } else {
+            isActive = true;
+            animateLine();
+        }
+    });
+
+    // requestAnimationFrame pauses once the tab becomes inactive
+    // reset startTime and progress once the tab is active again
+    document.addEventListener('visibilitychange', function() {
+        isActive = true;
```

Comment:
oh you may also want to add a condition here `if (document.visibilityState === 'visibile')`. Its not really necessary bc `requestAnimationFrame` gets paused when `visibilityState` is `hidden` but it might add some clarity. 💭